### DISCUSSION TITLE
music/hapax/

### DISCUSSION THREAD

**Discussion by giscus** [shane0/music#6] [General]
# music/hapax/

https://shanenull.com/music/hapax/

**Comment by shane0** [comment]
hapax DEFAULT is the name of the default project

hermod is _

**Reply by shane0** [reply]
# updates

- 2.13
- https://squarp.net/hapax/firmware/

**Comment by shane0** [comment]
https://squarp.net/hapax/manual/

**Reply by shane0** [reply]
# firmware 2.1

- https://youtu.be/sT7Hqh3Zje0

**Comment by shane0** [comment]
- fx pattern param locking = Hold  `2ND`  and Press the associated `encoder` to toggle the param locking on fx

**Comment by shane0** [comment]
- big video
- https://youtu.be/tTLQx2kexNM

**Reply by shane0** [reply]
Red https://youtu.be/uSYwI_TivGk?si=XJdSglciTNqjx_Dt

**Comment by shane0** [comment]
firmware 2.0

https://youtu.be/_vyLvKJpfGY

**Comment by shane0** [comment]
https://github.com/shane0/hapax-instruments
instruments file

**Comment by shane0** [comment]
insturment file editor
https://github.com/px-xp/track-definitions

**Comment by shane0** [comment]
# clock out

The Squarp Hapax is a powerful sequencer that offers flexible clock output options to synchronize your other gear. Here's a breakdown of how to configure it:

**1. MIDI Clock Output:**

* **Global MIDI Clock:** Hapax can send MIDI clock and transport (Start/Stop) messages from its dedicated MIDI outputs (MIDI A, MIDI B, USB HOST, USB DEVICE).
    * Go to **Settings > Sync Out**.
    * For each MIDI output you want to use, enable "CLOCK" to send BPM timing clock and "TRANSPORT" to send Start/Stop messages.
    * You can also choose to "SEND" clock even when Hapax is stopped, or only when it's playing ("CLOCK ON STOP: --").
* **Per-Track MIDI Output:** While the main clock is global, individual tracks can be routed to specific MIDI outputs and channels in the "Track settings menu" (hold a track pad). This allows you to sequence different instruments on different MIDI outputs.

**2. CV Clock Output:**

* Hapax has four CV/Gate outputs (CV1/GATE1, CV2/GATE2, CV3/GATE3, CV4/GATE4) that can be configured to output a clock signal. This is particularly useful for synchronizing Eurorack modules or other analog gear.
    * Go to **Settings > Sync Out**.
    * Under "GATE CLOCK," select the desired CV/Gate output (e.g., GATE 1).
    * Under "GATE CLOCK DIV," set the desired clock division (e.g., 1/16, 1/8, 1/4, 1/1).
    * You can also configure "GATE RUN" to send a continuous high/low signal when Hapax is playing/stopped, and "TRIG RESET" to send a brief trigger on start or stop.
* **DIN Sync:** Hapax also supports DIN Sync (Sync24) output, which can be sent from MIDI output C. This is useful for syncing vintage gear like Roland TR-808s or TR-909s. You can select the clock pulse resolution (e.g., 1/96 to 1/1).

**Important Considerations for Stable Clock Output:**

* **Hapax as Master:** For the tightest synchronization, it's often recommended to set the Hapax as the master clock source and slave your other devices to it.
* **MIDI Jitter:** While Hapax's MIDI clock is generally good, MIDI itself can introduce some jitter. For critical timing applications, an audio pulse clock from a DAW (fed into a Hapax CV input) is often recommended, as it provides extremely tight sync. In this scenario, Hapax would be slaved to the DAW's audio clock, and then Hapax can forward MIDI clock to other devices.
* **Cables and Connections:** Ensure you are using proper MIDI and CV cables and that connections are secure.
* **Compensation/Latency:** If you experience timing issues, especially when integrating with a DAW, explore latency compensation settings on both the Hapax and your DAW. Hapax allows for compensation per channel for notes and automation on MIDI outputs.
* **Output Levels (for CV):** When using CV clock, ensure the output levels from Hapax are sufficient for your receiving module (often close to 0dB, backing off slightly). For some audio interfaces being used to send clock to Hapax, they need to be DC-coupled.

By understanding these options and best practices, you can effectively use your Squarp Hapax to be the rhythmic brain of your studio. Always refer to the official Hapax manual for the most detailed and up-to-date information.

**Comment by shane0** [comment]
# track type

The Squarp Hapax is designed for incredible flexibility, and a key part of that is its different track types. You can easily switch a track between **Poly, Drum, or MPE** modes, fundamentally changing how the 128 pads and step sequencer behave for that track.

Here's how to set a Hapax track to Drum or Keys (Poly/MPE):

**The Core Method (Shortcut):**

The quickest way to change a track's type is:

1.  **Select the Track:** Press one of the 16 track buttons to select the track you want to change.
2.  **Hold SHIFT (or 2ND) + Rotate the Main Encoder:**
    * **Hold the `STEP` button.** (The `STEP` button acts as a shift modifier for this function).
    * **While holding `STEP`, rotate the main encoder.**
    * The screen will display the available track types: **POLY, DRUM, or MPE**. Select the desired type by rotating the encoder.

**Understanding the Track Types:**

* **POLY (Keys/Melody Tracks):**
    * This is the default track type.
    * The 128 pads act as a **chromatic or isomorphic keyboard**, allowing you to play melodies, chords, and polyphonic lines.
    * In Step mode, the grid acts as a **piano roll**, where rows represent notes and columns represent steps. This is ideal for traditional melodic and harmonic sequencing.
* **DRUM (Drum Machine Tracks):**
    * When you set a track to Drum mode, the 128 pads are organized into **8 drum lanes**, each with 16 velocity pads. This makes it incredibly intuitive for playing and programming drum beats.
    * Each drum lane can be assigned to a specific **MIDI note** and **MIDI channel**. This is crucial for triggering different drum sounds on your drum machine or sampler.
    * In Step mode, the grid changes to a **drum grid**, where each row represents a specific drum sound (kick, snare, hi-hat, etc.), and you can easily place steps for each instrument.
    * **Automation on Drum Tracks:** Drum tracks have a unique feature: their automation can be sent on a *separate MIDI channel* from the drum notes themselves. This is useful if your drum machine has a dedicated MIDI channel for effects or global parameters. You can configure this in the track's settings.
* **MPE (MIDI Polyphonic Expression Tracks):**
    * MPE tracks are designed for controllers that support MIDI Polyphonic Expression (MPE), such as the Expressive E Osmose, Roli Seaboard, or Sensel Morph.
    * This mode captures and sequences the individual expression data (pitch bend, pressure, timbre) for each note played, allowing for highly nuanced and expressive performances.
    * The pads also respond to MPE data, and the step sequencer can display and edit this rich per-note modulation.

**Configuring MIDI Outputs for Tracks:**

After setting the track type, remember to configure its MIDI output:

1.  **Hold the Track Pad:** Hold down the pad for the desired track (e.g., Track 1).
2.  **Track Settings Menu:** This opens the track's primary settings on the screen.
3.  **Output Port and Channel:**
    * **OUTPUT PORT:** Select the physical MIDI output (MIDI A, B, C, D, USB HOST, USB DEVICE, CV/GATE outputs) where your synth or drum machine is connected.
    * **OUTPUT CHANNEL:** Set the MIDI channel (1-16) that your connected instrument is listening on for that track.

By understanding these track types and how to switch between them, you can fully leverage the Hapax's power to sequence all kinds of instruments in your setup.

**Comment by shane0** [comment]
# troubleshooting midi in from hapax

- hapax midi c to black trs adapter to rample
- hapax set a track to dsrums track > step > right encoder
- track to midi channel 16
- notes 48 to 51 by holding the 8 encoders just used the bottom 4 since the rample is 4 tracks

Here's the markdown table showing Rample's default MIDI channel and note assignments for triggering the **first sample slot** on each track:

| Rample Track | MIDI Channel | Note Name | MIDI Note Number | Sample Slot Triggered |
|--------------|--------------|-----------|------------------|------------------------|
| **Track A**  | 1            | C1        | 36               | Slot 1                |
| **Track B**  | 2            | E2        | 52               | Slot 1                |
| **Track C**  | 3            | A#3       | 70               | Slot 1                |
| **Track D**  | 4            | E5        | 88               | Slot 1                |

### Key Details:
1. **Per-Track Channels**:  
   Each track requires a dedicated MIDI channel (Ch 1-4).

2. **Single-Note Trigger**:  
   Sending the specified note on the correct channel will play **Sample Slot 1** of that track.

3. **Full Note Ranges**:  
   Each track supports a range of notes for multiple samples. The table above shows only the first slot trigger.  
   Full ranges:
   - Track A: **C1 (36)** to D#2 (51) → *16 samples*
   - Track B: **E2 (52)** to A3 (69) → *16 samples*
   - Track C: **A#3 (70)** to D#5 (87) → *16 samples*
   - Track D: **E5 (88)** to C7 (96) → *8 samples*

### How to Use for Drum Pads:
- Program your Hapax to send:
  - **Note 36** on **Ch 1** → Triggers Track A (e.g., kick)
  - **Note 52** on **Ch 2** → Triggers Track B (e.g., snare)
  - **Note 70** on **Ch 3** → Triggers Track C (e.g., hat)
  - **Note 88** on **Ch 4** → Triggers Track D (e.g., crash)

> 💡 **Tip**: To trigger other sample slots, use adjacent notes (e.g., Note 37 = Slot 2 on Track A). See Rample's manual for full sample mapping.
